### PR TITLE
tests: fix TestAssetValidRounds

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

* The test validates first/last valid against possible expired value of lastRound.
  Make ComputeValidityRounds returning lastValid to check validity range against it.
* Fix some linter warnings
* Modify all usages of ComputeValidityRounds

## Test Plan

This is a test

### CODE REVIEW

File: test/e2e-go/features/transactions/asset_test.go [hunk 1]
```diff
@@ -80,45 +77,45 @@ func TestAssetValidRounds(t *testing.T) {
 	firstValid = 0
 	lastValid = 0
 	validRounds = cparams.MaxTxnLife + 1
-	firstValid, lastValid, err = client.ComputeValidityRounds(firstValid, lastValid, validRounds)
+	firstValid, lastValid, lastRound, err = client.ComputeValidityRounds(firstValid, lastValid, validRounds)
 	a.NoError(err)
-	a.Equal(params.LastRound+1, firstValid)
+	a.Equal(lastRound+1, firstValid)
```
Comment (nicholasguoalgorand) [inline test/e2e-go/features/transactions/asset_test.go:82]: what exactly changed here? Doesn't lastRound always get the value of params.LastRound?

Comment (algorandskiy) [reply test/e2e-go/features/transactions/asset_test.go:82]: params might be requested at round X but ComputeValidityRounds called at round X+1, so it returns the round it used as a calculation base

File: libgoal/libgoal.go [hunk 2]
```diff
@@ -513,17 +513,18 @@ func (c *Client) signAndBroadcastTransactionWithWallet(walletHandle, pw []byte,
 // 		 M     |     0     | first + validRounds - 1
 // 		 M     |     M     | error
 //
-func (c *Client) ComputeValidityRounds(firstValid, lastValid, validRounds uint64) (uint64, uint64, error) {
+func (c *Client) ComputeValidityRounds(firstValid, lastValid, validRounds uint64) (uint64, uint64, uint64, error) {
```
Comment (brianolson) [inline libgoal/libgoal.go:516]: it would be nice to have the return values named. `(firstValid, lastValid, previousRound, err)` ?

Comment (algorandskiy) [reply libgoal/libgoal.go]: done!

**Review by brianolson — APPROVED** [review]

**Review by jannotti — APPROVED** [review]
Looks like improvements to me.

I don't fully see what `ComputeValidityRounds` change is for. It seems like almost every caller is ignoring the new result. Maybe comment the one place that cares and explain why that callsite is being extra careful.

**Comment by algorandskiy** [comment]
before ComputeValidityRounds we had first valid and last valid computed in-place. So adding the function is a large improvement. But testing it is a problem since it calls algod to get data, and the caller did not have enough data to check its computation. Probably a right way of doing this is to spit the function and test the pure inner function. But I guess returning the base round from the calculation function is kind of useful.

**Pull request merged**